modules/fieldlinessequence/rendering/renderablefieldlinessequence.h

```diff
@@ -0,0 +1,175 @@
+    properties::Vec4Property     _pColorUniform;        // Uniform Field Line Color
+    properties::BoolProperty     _pColorABlendEnabled;  // Whether or not to use additive blending
+
+    properties::BoolProperty     _pDomainEnabled;       // Whether or not to use Domain
+    properties::PropertyOwner    _pDomainGroup;         // Group to hold the Domain properties
+    properties::Vec2Property     _pDomainX;             // Domain Limits along x-axis
+    properties::Vec2Property     _pDomainY;             // Domain Limits along y-axis
+    properties::Vec2Property     _pDomainZ;             // Domain Limits along z-axis
+    properties::Vec2Property     _pDomainR;             // Domain Limits radially
+
+    properties::Vec4Property     _pFlowColor;           // Simulated particles' color
+    properties::BoolProperty     _pFlowEnabled;         // Toggle flow [ON/OFF]
+    properties::PropertyOwner    _pFlowGroup;           // Group to hold the flow/particle properties
+    properties::IntProperty      _pFlowParticleSize;    // Size of simulated flow particles
+    properties::IntProperty      _pFlowParticleSpacing; // Size of simulated flow particles
+    properties::BoolProperty     _pFlowReversed;        // Toggle flow direction [FORWARDS/BACKWARDS]
+    properties::IntProperty      _pFlowSpeed;           // Speed of simulated flow
+
+    properties::BoolProperty     _pMaskingEnabled;      // Whether or not to use masking
+    properties::PropertyOwner    _pMaskingGroup;        // Group to hold the masking properties
+    properties::StringProperty   _pMaskingMin;          // Lower limit for allowed values
+    properties::StringProperty   _pMaskingMax;          // Upper limit for allowed values
+    properties::OptionProperty   _pMaskingQuantity;     // Index of the extra quantity to use for masking
+
+    properties::TriggerProperty  _pFocusOnOriginBtn;    // Button which sets camera focus to parent node of the renderable
+    properties::TriggerProperty  _pJumpToStartBtn;      // Button which executes a time jump to start of sequence
+
+    // --------------------- FUNCTIONS USED DURING INITIALIZATION --------------------- //
+    void   addStateToSequence(FieldlinesState& STATE);
+    void   computeSequenceEndTime();
+    void   definePropertyCallbackFunctions();
+    bool   extractJsonInfoFromDictionary(fls::Model& model);
+    bool   extractMandatoryInfoFromDictionary(SourceFileType& sourceFileType);
+    void   extractOptionalInfoFromDictionary(std::string& outputFolderPath);
+    void   extractOsflsInfoFromDictionary();
+    void   extractTriggerTimesFromFileNames();
+    bool   loadJsonStatesIntoRAM(const std::string& OUTPUT_FOLDER);
+    void   loadOsflsStatesIntoRAM(const std::string& OUTPUT_FOLDER);
+    void   setModelDependentConstants();
+    void   setupProperties();
+    bool   prepareForOsflsStreaming();
+#ifdef OPENSPACE_MODULE_KAMELEON_ENABLED
```

Comment:
Can these methods be moved out to a separate class? To make this class completely independent from Kameleon?